keras/applications/inception_v4.py

```diff
@@ -0,0 +1,321 @@
+    net = conv2d_bn(net, 64, 3, 3)
+
+    branch_0 = MaxPooling2D((3, 3), strides=(2, 2), padding='valid')(net)
+
+    branch_1 = conv2d_bn(net, 96, 3, 3, strides=(2, 2), padding='valid')
+
+    net = layers.concatenate([branch_0, branch_1], axis=channel_axis)
+
+    branch_0 = conv2d_bn(net, 64, 1, 1)
+    branch_0 = conv2d_bn(branch_0, 96, 3, 3, padding='valid')
+
+    branch_1 = conv2d_bn(net, 64, 1, 1)
+    branch_1 = conv2d_bn(branch_1, 64, 1, 7)
+    branch_1 = conv2d_bn(branch_1, 64, 7, 1)
+    branch_1 = conv2d_bn(branch_1, 96, 3, 3, padding='valid')
+
+    net = layers.concatenate([branch_0, branch_1], axis=channel_axis)
+
+    branch_0 = conv2d_bn(net, 192, 3, 3, strides=(2, 2), padding='valid')
+    branch_1 = MaxPooling2D((3, 3), strides=(2, 2), padding='valid')(net)
+
+    net = layers.concatenate([branch_0, branch_1], axis=channel_axis)
+
+    # 35 x 35 x 384
+    # 4 x Inception-A blocks
+    for idx in range(4):
+        net = block_inception_a(net)
+
+    # 35 x 35 x 384
+    # Reduction-A block
+    net = block_reduction_a(net)
+
+    # 17 x 17 x 1024
+    # 7 x Inception-B blocks
+    for idx in range(7):
+        net = block_inception_b(net)
+
+    # 17 x 17 x 1024
+    # Reduction-B block
+    net = block_reduction_b(net)
+
+    # 8 x 8 x 1536
+    # 3 x Inception-C blocks
+    for idx in range(3):
+        net = block_inception_c(net)
+    return net
+
+
+def InceptionV4(include_top=True,
+                weights="imagenet",
+                input_tensor=None,
+                input_shape=None,
+                classes=1000,
+                dropout_prop=0.2):
+    '''Instantiates the Inception V4 architecture.
+    '''
+
+    if weights not in {'imagenet', None}:
+        raise ValueError('The `weights` argument should be either '
+                         '`None` (random initialization) or `imagenet` '
+                         '(pre-training on ImageNet).')
+
+    if weights == 'imagenet' and include_top and classes != 1000:
+        raise ValueError('If using `weights` as imagenet with `include_top`'
+                         ' as true, `classes` should be 1000')
+
+    # Determine proper input shape
+    input_shape = _obtain_input_shape(
+        input_shape,
+        default_size=299,
+        min_size=139,
+        data_format=K.image_data_format(),
+        require_flatten=False,
+        weights=weights)
+
+    if input_tensor is None:
+        img_input = Input(shape=input_shape)
+    else:
+        if not K.is_keras_tensor(input_tensor):
+            img_input = Input(tensor=input_tensor, shape=input_shape)
+        else:
+            img_input = input_tensor
+
+    # Make inception base
+    x = inception_v4_base(img_input)
+
+    # Final pooling and prediction
+    if include_top:
+        # 1 x 1 x 1536
+        x = AveragePooling2D((8, 8), padding='valid')(x)
+        x = Dropout(dropout_keep_prob)(x)
+        x = Flatten()(x)
+        # 1536
+        x = Dense(units=num_classes, activation='softmax')(x)
```

Comment:
`num_classes` is also not defined

Reply:
uhhhhhhhhh I'm so stuppppppid :1st_place_medal: